mibidata/tiff.py

```diff
@@ -132,10 +139,10 @@ def write(filename, image, sed=None, optical=None, ranges=None,
                 max_value = (341, range_dtype, 1, ranges[i][1])
                 page_tags = coordinates + [page_name, min_value, max_value]
 
-                if write_float:
-                    to_save = image.data[:, :, i].astype(np.float32)
+                if range_dtype == 'I':
+                    to_save = image.data[:, :, i].astype(np.uint16)
                 else:
-                    to_save = image.data[:, :, i]
+                    to_save = image.data[:, :, i].astype(np.float)
```

Comment:
Here again, you're bypassing the choices you gave in the list and end up using only either `np.uint16` if it's integer type or `np.float` if it's float type.